### PR TITLE
Andi/farmediting2

### PR DESCRIPTION
**Pull request by andi-halim**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?

[//]: # '####### YOUR ANSWER BELOW ###########'
- Replace placeholders of FieldInput with the farm data that matches the record ID. 
- Editing should validate email address, mailing address, and phone numbers and display an error state (enabled via MUI) if not valid (see utils.js for field validation functions) 

### How to Review
- Check that grey placeholders match the Airtable data
- Check that incorrect values have errors. 
-

[//]: # 'The order in which to review files and 
what to expect when testing locally'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Related PRs

[//]: # "Optional - related PRs you're waiting on
/ PRs that will conflict, etc"
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Next steps

[//]: # "What doesn't work yet, what's NOT in this 
PR/has to be done "
[//]: # '####### YOUR ANSWER BELOW ###########'
I'm honestly still confused about Autocomplete. If it's as simple as setting something to true, maybe someone could look at my code and give that a go? 

### Screenshots

[//]: # '#### YOUR SCREENSHOTS BELOW ########'
image

### Design Status

[//]: # 'If this is a frontend PR, what is the expected 
status of the UI in this PR (lo, mid, high- fi?'
[//]: # '####### LOFI/MIDFI/HIFI ###########'
[//]: # '############## END ##################'

CC: @phoebeli23

### CODE REVIEW

File: src/screens/shared/farmProfileEdit/FarmProfileEditForm.js [hunk 1]
```diff
@@ -22,6 +30,20 @@ const styles = {
...
+    this.state = {
+      farm: {},
+      farmId: ''
+    };
+  }
+
+  async componentDidMount() {
+    const { farmId } = this.props;
+    const farm = await getSingleFarm(farmId);
+    this.setState({ farm, farmId });
```
Comment (mylesdomingo) [inline src/screens/shared/farmProfileEdit/FarmProfileEditForm.js:44]: Can you also call `handleChange` on farm in `componentDidMount` to update the parent state and see if it correctly updates state in `FarmProfileEdit`?

`handleChange(farm)`

File: src/screens/shared/farmProfileEdit/FarmProfileEditForm.js [hunk 2]
```diff
@@ -34,59 +56,65 @@ class FarmProfileEditForm extends React.PureComponent {
...
+              placeholder={farm.contactLastName}
               variant="outlined"
             />
           </div>
           <div className={classes.row}>
             <FieldInput
               label="Farm Name"
               onChange={this.onChange('farmName')}
-              placeholder="Andi's Potato Farm"
+              placeholder={farm.farmName}
```
Comment (mylesdomingo) [inline src/screens/shared/farmProfileEdit/FarmProfileEditForm.js]: can you change the placeholder prop here to be `value={farm.farmName}` etc for all farm props.

And replace the `placeholder={placeholder}` line 31 in `FieldInput`.

File: src/components/FieldInput.js [hunk 3]
```diff
@@ -17,18 +17,22 @@ const useStyles = makeStyles({
...
     <div className={classes.root}>
       {label ? <h3 className={classes.label}>{label}</h3> : null}
+
       <TextField
         fullWidth
         margin="normal"
         variant={variant || 'standard'}
         placeholder={placeholder}
         onChange={onChange}
+        {...(error && { error: true, helperText: error })}
```
Comment (mylesdomingo) [inline src/components/FieldInput.js]: do we need to shortcircuit this?

validateFunc gives '' = `false` if the field is valid, else a helper string which sets error to true. Also you already set helperText={error} below.

File: src/lib/utils.js [hunk 4]
```diff
@@ -113,6 +98,9 @@ const ValidateUSState = value => {
 
 // Ensure Zipcode is of valid length
 const validateZipcode = value => {
+  if (value === undefined) {
```
Comment (mylesdomingo) [inline src/lib/utils.js]: check if (!value) (for empty string)

File: src/screens/shared/farmProfileEdit/FarmProfileEditForm.js [hunk 5]
```diff
@@ -22,6 +30,20 @@ const styles = {
   }
 };
 class FarmProfileEditForm extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      farm: {},
```
Comment (mylesdomingo) [inline src/screens/shared/farmProfileEdit/FarmProfileEditForm.js:35]: We shouldnt create state here in the form. Use formValues given by props and use onChange to update the formValues object in the aprent component.

File: src/components/FieldInput.js [hunk 6]
```diff
@@ -17,18 +17,22 @@ const useStyles = makeStyles({
...
+        helperText={error}
+        autoComplete
```
Comment (mylesdomingo) [inline src/components/FieldInput.js]: sorry i was wrong 
autoComplete doesn't work right now but based on the [API ](https://material-ui.com/api/text-field/) you need to pass in a string of the right fiekd type (ie address, phone, etc). Refer to the docs.

File: src/screens/shared/farmProfileEdit/FarmProfileEditForm.js (hunk 2 above)
Comment (andi-halim) [reply src/screens/shared/farmProfileEdit/FarmProfileEditForm.js]: resolved.

File: src/screens/shared/farmProfileEdit/FarmProfileEditForm.js (hunk 5 above)
Comment (andi-halim) [reply src/screens/shared/farmProfileEdit/FarmProfileEditForm.js:35]: resolved.

File: src/lib/utils.js (hunk 4 above)
Comment (andi-halim) [reply src/lib/utils.js]: resolved.

**Pull request merged**